src/main/java/racingcar/step3/RacingCar.java

```diff
@@ -3,31 +3,52 @@
 import java.util.ArrayList;
 
 public class RacingCar {
-    private final int countOfCars;
+    private static final int MOVE_CONDITION = 4;
     private final int countOfRounds;
-    private int[] cars;
-    private ArrayList movementOfCar;
+    private ArrayList<Car> infoOfCars;
+    private ArrayList<Integer> movementOfCar;
 
-    public RacingCar(int countOfCars, int countOfRounds) {
-        this.countOfCars = countOfCars;
+    public RacingCar(String nameOfCars, int countOfRounds) {
+        this.infoOfCars = enterNameOfCars(nameOfCars);
         this.countOfRounds = countOfRounds;
-        this.cars = new int[countOfCars];
         this.movementOfCar = new ArrayList<>();
     }
 
-    public int getCountOfCars(){
-        return countOfCars;
+    public ArrayList<Car> getInfoOfCars() {
+        return infoOfCars;
     }
 
-    public int getCountOfRounds(){
+    public int getCountOfRounds() {
         return countOfRounds;
     }
 
-    public int[] getCars(){
-        return cars;
+    public int getCountOfCars() {
+        return infoOfCars.size();
     }
 
-    public ArrayList getMovementOfCar(){
+    public ArrayList<Integer> getMovementOfCar() {
         return movementOfCar;
     }
+
+    private ArrayList<Car> enterNameOfCars(String nameOfCars) {
+        String[] namesOfCar = nameOfCars.split(",");
+        infoOfCars = new ArrayList<>();
+        for (String carName : namesOfCar) {
+            infoOfCars.add(new Car(carName, 0));
+        }
+        return infoOfCars;
+    }
+
+    public void race(RacingCar racingCar) {
+        for (int i = 0; i < infoOfCars.size(); i++) {
+            racingCar.move(i, RandomNumber.makeRandomNumber());
```

Comment:
```suggestion
            move(i, RandomNumber.makeRandomNumber());
```
요렇게 사용하는건 안될까요?
인자로 나(Self)를 받아서 내 함수를 호출해야 할 이유가 있는지 궁금합니다🤔